autokeras/adapters/input_adapter.py

```diff
@@ -191,3 +191,78 @@ def infer_column_types(self):
         for key, value in column_types.items():
             if key not in self.column_types:
                 self.column_types[key] = value
+
+
+class TimeSeriesInputAdapter(adapter_module.Adapter):
+
+    def __init__(self,
+                 lookback=None,
+                 column_names=None,
+                 column_types=None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.lookback = lookback
+        self.column_names = column_names
+        self.column_types = column_types
+
+    def get_config(self):
+        config = super().get_config()
+        config.update({
+            'lookback': self.lookback,
+            'column_name': self.column_names,
+            'column_types': self.column_types
+        })
+        return config
+
+    def check(self, x):
+        """Record any information needed by transform."""
+        if not isinstance(x, (pd.DataFrame, np.ndarray, tf.data.Dataset)):
+            raise TypeError('Expect the data in TimeSeriesInput to be numpy.ndarray'
+                            ' or tf.data.Dataset or pd.DataFrame, but got {type}.'.
+                            format(type=type(x)))
+
+        if isinstance(x, np.ndarray) and x.ndim != 2:
+            raise ValueError('Expect the data in TimeSeriesInput to have 2 dimension'
+                             ', but got input shape {shape} with {ndim} '
+                             'dimensions'.format(
+                                 shape=x.shape,
+                                 ndim=x.ndim))
+
+        # Extract column_names from pd.DataFrame.
+        if isinstance(x, pd.DataFrame) and self.column_names is None:
+            self.column_names = list(x.columns)
+            # column_types is provided by user
+            if self.column_types:
+                for column_name in self.column_types:
+                    if column_name not in self.column_names:
+                        raise ValueError('Column_names and column_types are '
+                                         'mismatched. Cannot find column name '
+                                         '{name} in the data.'.format(
+                                             name=column_name))
+
+        # Generate column_names.
+        if self.column_names is None:
+            if self.column_types:
+                raise ValueError('Column names must be specified.')
+            self.column_names = [index for index in range(x.shape[1])]
+
+        # Check if column_names has the correct length.
+        if len(self.column_names) != x.shape[1]:
+            raise ValueError('Expect column_names to have length {expect} '
+                             'but got {actual}.'.format(
+                                 expect=x.shape[1],
+                                 actual=len(self.column_names)))
```

Comment:
Can we extract this part as a separate member function and let TimeSeriesInputAdapter to subclass StructuredDataInputAdapter? or we can extract a superclass for both of them to subclass.

Reply:
One issue is that the "update" function in the "StructuredDataInputAdapter" expects every row to be a vector of single values. But after the "convert_to_dataset" function ini "TimeSeriesInputAdapter" the dataset is transformed with every row to be a vector of vectors. This causes the "decode" function to fail. 

Also, the concept of column_names loses it's meaning after the data transform , so is it really relevant information that needs to be kept in the class?

Reply:
You can use whatever way you think makes the most sense to reuse the this part of check code. Superclass is one way, or you can use a stand alone-function.

Yes, we need to keep it. The user may want to access it later.

Reply:
Alright for now I've used reused the check function. Since that is the only thing we really need.